Remove use-database statement check

aswasif007 opened the pull request:
## Description

Currently, we have check for the `USE DATABASE` statement for `vip dev-env import sql` commands, though we don't have it for `vip import sql` command. In this PR, we are removing that check.

## Steps to Test

1. Check out PR.
2. Run `rm -rf dist && npm link`
3. Run `node ./dist/bin/vip dev-env create` to create a new dev environment with slug `vip-local`. You may have to do `export VIP_PROXY=socks://localhost:8080` if you are an Automattician and using proxy.
4. Run `node ./dist/bin/vip dev-env import sql --slug vip-local __fixtures__/validations/bad-sql-dev-env.sql`
5. The output should not contain the following lines:
```
Error:  SQL Error: USE <DATABASE_NAME> statement on line(s) 4.
Recommendation: Remove these lines
```
6. Also ensure that, `CREATE` and `USE DATABASE` statements have been removed from the file `__fixtures__/validations/bad-sql-dev-env.sql`

aswasif007 committed:
Remove use-database statement check

For the dev-env-import-sql command, we are stripping the check for use-database statements

aswasif007 committed:
Remove comment that is not relevant anymore

abdullah-kasim commented:
Slack discussion: https://a8c.slack.com/archives/CLKJLBFB8/p1674116212483679

Adding these as I move around as I've greatly benefitted from looking through git history for a specific line to see why it was written as such.

Maybe I'll see this again in years to come :)

aswasif007 committed:
Remove USE and CREATE DATABASE statements from the sql file

aswasif007 committed:
Address a security issue

aswasif007 commented on src/bin/vip-dev-env-import-sql.js:
```diff
@@ -64,6 +65,9 @@
 			const resolvedPath = await resolveImportPath( slug, fileName, searchReplace, inPlace );
 
+			// Remove the `USE` and `CREATE DATABASE` statements from the SQL file.
+			ChildProcess.execSync( `sed --in-place -E '/^(USE|CREATE DATABASE)/Id' ${ resolvedPath }` );
```
Thank you good bot. Addressed: https://github.com/Automattic/vip-cli/pull/1253/commits/04d6bd392ec4882206c75dbaeb6a424f69246947

saroshaga commented on src/bin/vip-dev-env-import-sql.js:
```diff
@@ -64,6 +65,12 @@ command( {
 
+			// Remove the `USE` and `CREATE DATABASE` statements from the SQL file.
+			ChildProcess.execFileSync(
+				'sed',
```
Would this work always or would you have to provide the path to the actual executable file? `

aswasif007 replied:
`sed` is a standard linux command, so this should work out of the box.

aswasif007 replied:
Ah.... this will be executed on user's OS. And `sed` is not available on windows. :grimacing:

saroshaga replied:
yup 😄

aswasif007 replied:
On another thought, we don't support vip-cli on windows natively. One has to install windows subsystem for linux to be able to vip-cli on windows: https://docs.wpvip.com/technical-references/vip-cli/installing-vip-cli/

And `sed` is available on WSL.

aswasif007 committed:
Refactor getReadAndWriteStreams

This is done to extract the temp file creation logic in a separate function: copyToTempFile

aswasif007 committed:
Consistent output from resolveImportPath

Irrespectively of whether search-replace was done or not, if inPlace is false, we want this function to return a temporary file

aswasif007 committed:
Trim unnecessary logic

aswasif007 committed:
Fix lint error

aswasif007 committed:
Fix tests
